Documentation/InjectionPatterns.md

```diff
@@ -68,39 +68,39 @@ class PetOwner2: Person {
 }
 ```
 
-Or, you can use <a name="initialization-callback">`initCompleted` callback</a> instead of defining the injection in the registration closure:
+Note that the <a name="initialization-callback">`initCompleted` callback</a> in some cases is the same as defining the injection in the registration closure (see the following example), but `initCompleted` is preferred since it is able to resolve _[circular dependencies](CircularDependencies.md)_.
```

Comment:
It is true that it is necessary for circular dependencies, but i wouldn't say that it is therefore prefered - it often requires type casting, which looses type safety

Reply:
I know this is out of the scope of this PR, but is it possible to use generics to have the correct type as a parameter in the initCompleted closure?

Reply:
It is quite problematic - mostly because you cannot require relationships between generic parameters: Something like
```
func foo<T, U>(type: T.Type) -> U where U: T
```
is not possible